reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/CompositeDisposable.kt

```diff
@@ -26,8 +26,11 @@ class CompositeDisposable : Disposable {
         set
             .getAndUpdate {
                 it
-                    ?.filterNotTo(hashSetOf(), Disposable::isDisposed)
-                    ?.plus(disposable)
+                    ?.toCollection(ArrayList(it.size + 1))
+                    ?.apply {
+                        removeAll(Disposable::isDisposed)
```

Comment:
It will work because this `ArrayList` is not frozen yet, right?

Reply:
Yep